### DISCUSSION TITLE
Bug: Support for 18.16.0 was dropped

### DISCUSSION THREAD

**Discussion by sliftist** [eslint/eslint#19057] [General]
It looks like support for NodeJS 18.16.0 was dropped, perhaps unintentionally. Change was made here https://github.com/eslint/eslint/commit/e1e827ffcbd73faa40dbac3b97529452e9c67108, per ticket https://github.com/eslint/eslint/issues/17595 , with changes mentioning dropping support for 12, 14, 16, 17, 19.

This changes means I have to update machines from 18.16.0, which does not appear to be the intent of the original change.

Would a pull request to change the node dependency from `"node": "^18.18.0 || ^20.9.0 || >=21.1.0"` to `"node": "~18.18.0 || ~20.9.0 || >=21.1.0"` be accepted, or is there a specific reason 18.16.0 was removed?

**Comment by fasttime** [comment]
> It looks like support for NodeJS 18.16.0 was dropped, perhaps unintentionally. Change was made here [e1e827f](https://github.com/eslint/eslint/commit/e1e827ffcbd73faa40dbac3b97529452e9c67108), per ticket [#17595](https://github.com/eslint/eslint/issues/17595) , with changes mentioning dropping support for 12, 14, 16, 17, 19.
> 
> This changes means I have to update machines from 18.16.0, which does not appear to be the intent of the original change.

Hi @sliftist! This is not a bug. If you scroll down on issue #17595 you can see that dropping support for Node.js < 18.18.0 was part of the discussion, even if it's not mentioned in the original proposal: https://github.com/eslint/eslint/issues/17595#issuecomment-1789192287. It's a completely intentional change.

> Would a pull request to change the node dependency from `"node": "^18.18.0 || ^20.9.0 || >=21.1.0"` to `"node": "~18.18.0 || ~20.9.0 || >=21.1.0"` be accepted, or is there a specific reason 18.16.0 was removed?

Node.js 18 had already reached [maintenance status](https://github.com/nodejs/Release/blob/1f4ae6bdfbe4ea77a6a85b42511ce97d517270c1/schedule.json#L98-L104) at the time of that post, with Node.js 18.18.0 being the latest minor release in the line. That is why support for older versions was removed. I doubt that a request to restore support would be accepted by the team at this time. I'd encourage you to upgrade your setup.

**Reply by jgawrych-echo360** [reply]
I disagree with this reasoning. `engine` is not a setting you can override. Requiring "the latest minor patch version at the time of the commit" without the package having any need on that minor version means that in order to use the latest eslint, we have to upgrade our node version. That may be trivial for a small personal project, but a company running dozens of microservices, across many environments, using runners on a minor outdated version of node, it can be a large undertaking. The only other option is to completely disable the safeguards by ignoring engines entirely.

I hope that eslint would reconsider lessening their engine requirement. I think it would be sufficient to say `^20 || ^22 || >=24` (honestly even just `>=20.0.0`), unless there's an actual feature or bug in node that eslint requires to operate.
